### PR TITLE
TOC, reference/tools: add a glossary for DM

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What is changed, added or deleted? 

add a glossary for DM

### Which version does your change affect? 

"dev", I'll update "v3.0"/"v3.1" after received two LGTM.

### CODE REVIEW

**Comment by WangXiangUSTC** [comment]
LGTM

**Comment by WangXiangUSTC** [comment]
how about add `safe-mode`, `checkpoint`, or we can add more glossary later

**Comment by csuzhangxc** [comment]
> how about add `safe-mode`, `checkpoint`, or we can add more glossary later

@WangXiangUSTC added `safe-mode` and `checkpoint` in https://github.com/pingcap/docs-cn/pull/2066/commits/face68c90e29099e94f230e8016eff20cc8c42fc.

File: dev/reference/tools/data-migration/glossary.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+### Shard Group
+
+指合库合表迁移过程中，需要合并迁移到下游同一张表的所有上游分表（shards），DM 内部具体实现时使用了两级抽象的 Shard Group，具体可查看[分库分表合并同步实现原理](/dev/reference/tools/data-migration/features/shard-merge.md#实现原理)。在当前文档中，有时也称作 Sharding Group。
+
+### Subtask
+
+数据迁移子任务，即数据迁移任务运行在单个 DM-worker 实例上的部分。根据任务配置的不同，单个数据迁移任务可能只有一个子任务，也可能有多个子任务。
+
+### Subtask Status
```
Comment (CaitinChen) [inline dev/reference/tools/data-migration/glossary.md]:
```suggestion
### Subtask status
```

File: dev/reference/tools/data-migration/glossary.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+### Table routing
+
+用于支持将上游 MySQL/MariaDB 实例的某些表同步到下游指定表的路由功能，可以用于分库分表的合并同步，具体可参考 [Table routing](/dev/reference/tools/data-migration/features/overview.md#table-routing)。
+
+### Task
+
+数据迁移任务，执行 `start-task` 命令成功后即启动一个数据迁移任务。根据任务配置的不同，单个数据迁移任务既可能只在单个 DM-worker 实例上运行，也可能同时在多个 DM-worker 实例上运行。
+
+### Task Status
```
Comment (CaitinChen) [inline dev/reference/tools/data-migration/glossary.md]:
```suggestion
### Task status
```

File: dev/reference/tools/data-migration/glossary.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+数据迁移子任务所处的状态，由 [Subtask Status](#subtask-status) 整合而来，具体信息可查看[任务状态](/dev/reference/tools/data-migration/query-status.md#任务状态)。
```
Comment (CaitinChen) [inline dev/reference/tools/data-migration/glossary.md]:
```suggestion
数据迁移子任务所处的状态，由 [Subtask status](#subtask-status) 整合而来，具体信息可查看[任务状态](/dev/reference/tools/data-migration/query-status.md#任务状态)。
```

**Comment by csuzhangxc** [comment]
@CaitinChen all comments addressed, PTAL again, thanks a lot!

**Comment by CaitinChen** [comment]
LGTM. Please also update the other versions. Thanks.

**Comment by csuzhangxc** [comment]
> LGTM. Please also update the other versions. Thanks.

@CaitinChen updated v3.0 and v3.1 in https://github.com/pingcap/docs-cn/pull/2066/commits/e1a07bc3c04981a9c126e3a5f2350d577bdbb9cd.

**Comment by CaitinChen** [comment]
@TomShawn Please help create English documents accordingly. Thanks.

**Pull request merged**